cpp/src/detail/shuffle_wrappers.cu

```diff
@@ -335,5 +335,103 @@ template rmm::device_uvector<size_t> groupby_and_count_edgelist_by_local_partiti
   std::optional<rmm::device_uvector<double>>& d_edgelist_weights,
   bool groupby_and_counts_local_partition);
 
+template <typename vertex_t, typename value_t, bool multi_gpu>
+void collect_vertex_values_to_local(raft::handle_t const& handle,
+                                    rmm::device_uvector<vertex_t>& d_vertices,
+                                    rmm::device_uvector<value_t>& d_values,
+                                    rmm::device_uvector<value_t>& d_local_values,
+                                    vertex_t local_vertex_first)
```

Comment:
This looks same to the code added in PR #2150 and the same comment applies here.